index.js

```diff
@@ -340,11 +340,11 @@ var Base = Class.extend({
 
     var columnNameArray = {};
 
-    if( arguments.length > 3 ) {
+    if( arguments.length > 3 || Array.isArray(callback)) {
 
       columnNameArray = valueArray;
       valueArray = callback;
-      callback = arguments[3];
+      callback = arguments[3] ? arguments[3] : throw;
```

Comment:
What should this line accomplish? I don't think it is a good idea to put in throw on missing callbacks. That will definitely collide with promises.

Reply:
Ah, I see, it puts the callback into nodeify at the end. In that case I can check that callback is defined before trying to use it on line 336. Should I amend this commit or create a new one?

Reply:
as you want, we don't enforce squashing though.